java/src/com/ibm/streamsx/topology/internal/context/BuildServiceCompile.java

```diff
@@ -0,0 +1,145 @@
+        
+        return new CompletedFuture<File>(bundle);
+	}
+	
+	@Override
+	public Future<File> submit(JSONObject submission) throws InterruptedException, ExecutionException, Exception{
+		JSONObject jsonGraph = (JSONObject) submission.get(SUBMISSION_GRAPH);
+		String namespace = jsonGraph.get("namespace").toString();
+		String mainComposite = jsonGraph.get("name").toString();
+		
+		File zippedArchive = super.submit(submission).get();
+		File bundle = doRemoteCompile(namespace, mainComposite, zippedArchive);
+
+		return new CompletedFuture<File>(bundle);
+		
+	}
+	
+	private File doRemoteCompile(String namespace, String mainComposite, 
+			File zippedArchive) throws ClientProtocolException, IOException {
+		String bundleName = namespace + "." + mainComposite + ".sab";
+		String archiveName = zippedArchive.getName();
+		HttpClient client = HttpClientBuilder.create().build();
+		
+		File archiveFile = new File(archiveName);
+		HttpPost postJobWithConfig = new HttpPost(getBuildFarmURL());
+        postJobWithConfig.addHeader("accept", ContentType.APPLICATION_JSON.getMimeType());
+
+        FileBody bundleBody = new FileBody(archiveFile, 
+        		ContentType.APPLICATION_OCTET_STREAM, archiveName);
+        
+        StringBody configBody = new StringBody(createBuildConfig(namespace, mainComposite),
+        		ContentType.TEXT_PLAIN);
+
+        HttpEntity reqEntity = MultipartEntityBuilder.create()
+                .addPart("inputfile", bundleBody).addPart("buildconfig", configBody)
+                .build();
+
+        postJobWithConfig.setEntity(reqEntity);
+        
+        File bundle = doPostAndRetreiveFile((CloseableHttpClient) client, 
+        		postJobWithConfig, bundleName);       
+        return bundle;
+	}
+
+	private String getBuildFarmURL(){
+		// For the moment, just fetch a static public build farm url. 
+		// TODO: grab from VCAP file
+		return "https://streamsbuildservice.mybluemix.net/api/build/compile/spltoolkit";
+	}
+
+	private String createBuildConfig(String namespace, String mainComposite){
+		String config = "{"
+							+ "\"namespace\":\"" + namespace + "\","
+							+ "\"mainComposite\":\"" + mainComposite + "\""
+							+ ",\"streamsVersion\":\"4.2\""
+							+ ",\"pythonVersion\":\"3.5\""
+							+ ",\"timeout\":100"
+					  + "}";
+		return config;
+	}
+	
+	
+    private File doPostAndRetreiveFile(CloseableHttpClient httpClient,
+            HttpRequestBase request, String bundleName) 
+            		throws ClientProtocolException, IOException {
+        request.addHeader("accept",
+                ContentType.APPLICATION_JSON.getMimeType());
+
+        File bundle;
+        CloseableHttpResponse response = AnalyticsServiceStreamsContext
+        		.executeAndHandleErrors(httpClient, request);
+        HttpEntity entity = response.getEntity();
+        
+        try { 	
+            InputStream is = null;
+            FileOutputStream fos = null;
+            byte[] buffer = new byte[1024];
```

Comment:
What size would you recommend? I picked it somewhat arbitrarily. The sab files we're retrieving can be anywhere from 1-300 MB (largest I've heard of).